compiler/src/main/scala/org/scalajs/core/compiler/JSGlobalAddons.scala

```diff
@@ -95,6 +95,12 @@ trait JSGlobalAddons extends JSDefinitions
       sys.error("non-exported name passed to jsInfoSpec")
     }
 
+    def registerJSExposedSymbol(sym: Symbol): Unit =
+      sym.addAnnotation(ExposedJSMemberAnnot)
+
+    def isExposed(sym: Symbol): Boolean =
+      sym.hasAnnotation(ExposedJSMemberAnnot)
+
```

Comment:
I am assuming you need to do this, because in the backend you are unable to determine the orignal privateness reliably. If the privateness can be determined, we should not expose this in the frontend (or is this a binary compatibility concern)?

Further, I would refrain from using the name `register` here, since it is used differently in the same context (for per-run information). This should follow a equivalent path then adding and testing for the `@RawJSType` annotation. (Which is currently not abstracted but directly used in place).

I propose to either:

1. Remove and inline these two methods at usage site (or declare helpers in the relevant phases)
2. Change the name register, and change the `@RawJSType` logic to use the same flow.

I prefer 1, mainly because it allows us to have less stuff which is shared globally among phases.